src/unit/test_hashset.c

```diff
@@ -371,6 +371,67 @@ int test_scan(int argc, char **argv, int flags) {
     return 0;
 }
 
+void scanrehashfn(void *privdata, void *element) {
+    scandata *data = (scandata *)privdata;
+    unsigned long j = (unsigned long)element;
+    if (j < (unsigned long)data->count) {
+        data->element_seen[j]++;
+    }
+}
+
+int test_scan_with_rehash(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    int num_rounds = (flags & UNIT_TEST_ACCURATE) ? 1000 : 100;
+    int num_elements_inspected = 100;
+
+    /* A set of longs, i.e. pointer-sized values. */
+    hashsetType type = {0};
+    long j;
+
+    for (int round = 0; round < num_rounds; round++) {
+        const long count = num_elements_inspected + 1 + (rand() % 1000);
+
+        /* Seed, to make sure each round is different. */
+        test_set_hash_function_seed(argc, argv, flags);
```

Comment:
I created a helper for this.

```suggestion
        randomSeed();
```